pkg/mcs/resourcemanager/server/manager.go

```diff
@@ -296,52 +310,52 @@ func (m *Manager) AddResourceGroup(grouppb *rmpb.ResourceGroup) error {
 // ModifyResourceGroup modifies an existing resource group.
 func (m *Manager) ModifyResourceGroup(grouppb *rmpb.ResourceGroup) error {
 	keyspaceID := ExtractKeyspaceID(grouppb.GetKeyspaceId())
-	krgm := m.getKeyspaceResourceGroupManager(keyspaceID)
-	if krgm == nil {
-		return errs.ErrKeyspaceNotExists.FastGenByArgs(keyspaceID)
+	krgm, err := m.accessKeyspaceResourceGroupManager(keyspaceID, grouppb.Name)
+	if err != nil {
+		return err
 	}
 	return krgm.modifyResourceGroup(grouppb)
 }
 
 // DeleteResourceGroup deletes a resource group.
 func (m *Manager) DeleteResourceGroup(keyspaceID uint32, name string) error {
-	krgm := m.getKeyspaceResourceGroupManager(keyspaceID)
-	if krgm == nil {
-		return errs.ErrKeyspaceNotExists.FastGenByArgs(keyspaceID)
+	krgm, err := m.accessKeyspaceResourceGroupManager(keyspaceID, name)
```

Comment:
NIT: "default" group can't be deleted, so maybe there is not need to call `accessKeyspaceResourceGroupManager`